packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue

```diff
@@ -0,0 +1,408 @@
+            this.$log.error('Error getting self exclusion status', error);
+        }
+    },
+
+    beforeCreate () {
+        if (!this.$store.hasModule('fSelfExclusionModule')) {
+            this.$store.registerModule('fSelfExclusionModule', fSelfExclusionModule);
+        }
+    },
+
+    methods: {
+        ...mapActions('fSelfExclusionModule', [
+            'getExclusions',
+            'updateAlcoholExclusion'
+        ]),
+
+        formatDate (date) {
+            const dd = String(date.getDate()).padStart(2, '0');
+            const mm = String(date.getMonth() + 1).padStart(2, '0');
+            const yyyy = date.getFullYear();
+            return `${dd}-${mm}-${yyyy}`;
+        },
+
+        changeState (state) {
+            this.selectedState = state;
+            this.closeAlertConfirmation();
+        },
+
+        async submitExclusionStatus () {
+            const api = this.selfExclusionApi;
+            const token = this.authToken;
+            const exclusionState = this.selectedState;
+
+            try {
+                await this.updateAlcoholExclusion({
+                    api,
+                    token,
+                    exclusionState
+                });
+
+                this.$log.info('Alcohol Exclusion saved successfully');
+                this.openAlertSuccess();
+            } catch (error) {
+                this.$log.error('Error saving consumer details', error);
+                this.openAlertError();
+            }
+        },
+
+        openAlertSuccess () {
+            this.closeAllAlerts();
+            this.isOpenAlertSuccess = true;
+        },
+
+        openAlertError () {
+            this.closeAllAlerts();
+            this.isOpenAlertError = true;
+        },
+
+        openAlertConfirmation () {
+            this.closeAllAlerts();
+            this.isOpenAlertConfirmation = true;
+        },
+
+        closeAlertConfirmation () {
+            this.isOpenAlertConfirmation = false;
+        },
+
+        closeAllAlerts () {
+            this.closeAlertConfirmation();
+        },
+
+        cancelLeave () {
+            this.$emit('cancel-leave');
+        },
+
+        confirmLeave () {
+            this.$emit('confirm-leave');
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use "@justeat/fozzie/src/scss/fozzie" as f;
+
+.c-selfExclusion-card-component > div {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+}
+
+.c-selfExclusion-title {
+    @include f.font-size(heading-m);
+}
+
+.c-selfExclusion-details,
+.c-selfExclusion-subtitle {
+    @include f.font-size(heading-s);
+}
+
+.c-selfExclusion-details {
+    margin-top: f.spacing(c);
+    margin-bottom: f.spacing(b);
+}
+
+.c-selfExclusion-fieldset {
+    display: flex;
+    flex-flow: column;
+    border: none;
+    padding: 0;
+}
+
+.c-selfExclusion-fieldset-date {
+    margin-left: f.spacing(f);
+    opacity: 0.5;
+}
+
+.c-selfExclusion-bottom-sheet-container {
```

Comment:
Just to fit with our component naming scheme:

```suggestion
.c-selfExclusion-bottomSheet {
```